lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1277,8 +1277,17 @@ void AttributeChecker::visitSPIAccessControlAttr(SPIAccessControlAttr *attr) {
     // VD must be public or open to use an @_spi attribute.
     auto declAccess = VD->getFormalAccess();
     auto DC = VD->getDeclContext()->getAsDecl();
-    if (declAccess < AccessLevel::Public &&
-        !VD->getAttrs().hasAttribute<UsableFromInlineAttr>() &&
+
+    AbstractStorageDecl *storage = nullptr;
+    if (auto *AD = dyn_cast<AccessorDecl>(VD))
+      storage = AD->getStorage();
+
+    auto canUseAttr = [](ValueDecl *VD) {
+      return VD->getFormalAccess() >= AccessLevel::Public ||
+        VD->getAttrs().hasAttribute<UsableFromInlineAttr>();
```

Comment:
Should this be VD->isUsableFromInline() which checks for other inlinable attributes?

Reply:
That could be worthwhile but I think it would allow more uses of `@_spi`. At this point this PR only aims to preserve the existing behavior and avoid breaking swiftinterface compatibility.